### PR TITLE
Fix issue #5: add package.json and .npmignore

### PR DESCRIPTION
**Pull request by Elchi3**
Requesting super review:
@teoli2003 
@jwhitlock 
@SebastianZ 
@wbamberg

### CODE REVIEW

**Comment by iamstarkov** [comment]
can i help to get it out?

**Comment by Elchi3** [comment]
> @Elchi3, I'm sorry I lost track of this request. I'm not a npm package developer, so I'm basing my review on the package.json docs and caniuse-db's package.json. 

No worries, I actually don't know who could review this, so I picked random team members.

> Based on the docs, I think the license should be MPL-2.0.

OK, I will update this PR.

> I'm not sure if they will allow us to publish without an author directive. If not, I am OK with @Elchi3 listed as the author, and add an AUTHORS file to populate the contributors. We can wait until 1.0.1 for that. Or skip right to 1.0.10000001, to match [caniuse's version scheme].

There is `"author": "Mozilla Developer Network"` currently, is this not sufficient? 

> can i help to get it out?

I still think issue #40 would be good to have fixed before releasing.

**Comment by jwhitlock** [comment]
>> I'm not sure if they will allow us to publish without an author directive. If not, I am OK with @Elchi3 >> listed as the author, and add an AUTHORS file to populate the contributors. We can wait until 1.0.1 >> for that. Or skip right to 1.0.10000001, to match [caniuse's version scheme].

> There is "author": "Mozilla Developer Network" currently, is this not sufficient?

Yes, I missed that. The author directive of just a name should work, since [email and URL are optional](https://docs.npmjs.com/files/package.json#people-fields-author-contributors).  NPM user info will populate a "maintainers" directive.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    "url": "https://github.com/mdn/data.git"
+  },
+  "keywords": [
+    "data",
+    "mdn",
+    "mozilla",
+    "css"
+  ],
+  "author": "Mozilla Developer Network",
+  "license": "MPL 2.0",
```
Comment (Elchi3) [inline package.json]: Fixed.

**Comment by iamstarkov** [comment]
are we good to go?

**Comment by iamstarkov** [comment]
@Elchi3 @jwhitlock can we merge this and publish to npm?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "mdn-data",
+  "version": "1.0.0",
+  "description": "Open Web data by the Mozilla Developer Network",
+  "repository": {
```
Comment (iamstarkov) [inline package.json]: i would say add `main` entrypoint field, which will point to `index.js`, which in turn will expose all json files from `./css/` folder.

Comment (iamstarkov) [inline package.json]: I made a PR #51 from this suggestion

**Comment by Elchi3** [comment]
Thanks for your interest in the npm package. As said earlier, issue #40 would be good to have before we release as it changes the data quite substantially (imo). Myself and other maintainers of this repository are busy with other things right now, so you'll need some more patience. Thanks again!

**Comment by iamstarkov** [comment]
@Elchi3 right, i got that you are busy and you want to get other things to be merged:
1) #40 is a good and can be released in upcoming version update. 
2) can i help you with github/npm release bureaucracy?

Right now problem with this repo is that one cant simply use it in nodejs env, because there is no `package.json` in github repo; if repo would have one (event not being published to npm), then this repo will be consumable by nodejs projects ([`npm i -D mdn/data#master`](https://docs.npmjs.com/cli/install))


> A package is:
>    a) a folder containing a program described by a package.json file
>    …
> Even if you never publish your package, you can still get a lot of benefits of using npm if you just want to write a node program (a), and perhaps if you also want to be able to easily install it elsewhere after packing it up into a tarball (b).
> 
>     npm install githubname/reponame


Regarding waiting for #40 before merging this PR, i would say its redundant (excessive?) requirement, because fixes are supposed to be released in upcoming releases, right?

If i were you, i would do things in this order:

1) merge #51 to get proper fs agnostic npm package
2) merge #36 to get npm package in master
3) add CHANGELOG for initial relase
4) go with minor release: `$ npm version minor && npm publish` (minor because #40 is kind of breaking change, so you dont want to have 2.0.0 as a second release)

then I would wait for #40 issue to be fixed, and then release it as another minor release, under 0.x.x its okay accordingly to semver.

From my point of view these things are quite not time-consuming, but anyway i can offer my time to help out

**Pull request merged**